src/main/scala/firrtl/Emitter.scala

```diff
@@ -22,10 +23,147 @@ import scala.collection.mutable.{ArrayBuffer, LinkedHashMap, HashSet}
 
 case class EmitterException(message: String) extends PassException(message)
 
-class FirrtlEmitter extends Emitter {
+// ***** Annotations for telling the Emitters what to emit *****
```

Comment:
I think its time to split Emitter into separate files, and probably rework the directory structure. This can wait for another PR, though.